### PR TITLE
Preliminary algorithm for direct mapping of keys to names.

### PR DESCRIPTION
**Pull request by abhinavmsra**
-> An initial layout of faker engine.
-> Maps the received key to appropriate dictionary field.

### CODE REVIEW

File: js/faker.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  * */
+  this.fetch = function (key) {
+    var penetrationDepth = $dictionaryRef + key;
+    var applicableDomain = eval(penetrationDepth);
+
+    if (!applicableDomain) { // exists in dictionary
+      return customExtraction(key);
+    }
+    else {
+      return applicableDomain[getRandomArbitrary(applicableDomain.length - 1, 0)];
```
Comment (ojash) [inline js/faker.js]: Try to maintain constants.

File: js/faker.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  *
+  * @return [String], bestMatch to fill the form
+  * */
+  var customExtraction = function (key) {
+    var bestMatch = [];
+    switch (key) {
+      case 'name.fullName':
+        Object.keys($.fakifyDictionary.name).forEach(function (index) {
+          var keyDomain = $.fakifyDictionary.name[index];
+          var seedIndex = getRandomArbitrary(keyDomain.length - 1, 0);
```
Comment (ojash) [inline js/faker.js]: Please do try to maintain constants.

File: js/faker.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+   *
+   * @return [Integer], a pseudo-random integer within the desired range
+   * */
+  var getRandomArbitrary =  function(max, min) {
+    return Math.floor((Math.random() * max) + min);
+  }
+
+}
+
+
```
Comment (ojash) [inline js/faker.js]: Maintain a single blank line at the end of the file

**Comment by abhinavmsra** [comment]
:ok:

**Comment by anantalft** [comment]
:+1:

**Pull request merged**